Sources/BottomSheetCalculator.swift

```diff
@@ -30,68 +30,98 @@ struct BottomSheetCalculator {
         return max(superview.frame.height - makeTargetHeight(), handleHeight)
     }
 
-    /// Calculates bottom and top thresholds for the given target offsets.
+    /// Creates the layout of the BottomSheetView based on the target offsets and the current target offset
+    ///
     /// - Parameters:
-    ///   - contentView: the content view of the bottom sheet
-    ///   - superview: the bottom sheet container view
-    ///   - height: preferred height for the content view
-    static func thresholds(for targetOffsets: [CGFloat], in superview: UIView) -> [CGFloat] {
+    ///   - targetOffsets: array containing the different target offsets a BottomSheetView can transition between
+    ///   - currentTargetIndex: index of the current target offset of the BottomSheetView
+    ///   - isDismissable: flag specifying whether the last two offsets should dismiss the BottomSheetView
+    static func createLayout(for targetOffsets: [CGFloat], at currentTargetIndex: Int, isDismissible: Bool) -> [BottomSheetModel] {
```

Comment:
As we discussed, it's not necessary to pass `isDismissible` flag in here. We can always mark the model as dismissible if it is and then call the delegate in `BottomSheetView`.